pyrival/algebra/discrete_log.py

```diff
@@ -1,22 +1,50 @@
+from math import gcd, sqrt
+
+
+def extended_gcd(a, b):
+    if not b:
+        return a, 1, 0
+
+    g, y, x = extended_gcd(b, a % b)
+    return g, x, y - ((a // b) * x)
```

Comment:
Please use, https://github.com/cheran-senthil/PyRival/blob/master/pyrival/algebra/gcd.py#L11 for extended_gcd.

Reply:
I'm assuming you mean to import that function and not copy-pasting the exact function used?

I actually considered doing that initially. The reason I did not do so is because in some places like the one below, I've seen existing methods being re-implemented instead of importing. https://github.com/cheran-senthil/PyRival/blob/1971590e96b351288cacb09852896f2383862ab6/pyrival/linear_algebra/multivariable_crt.py#L8 

I'm kinda confused about what pattern we're trying to follow here in this code base. Re-use existing methods whenever possible by importing or keep modules less tightly coupled allowing some duplication. Come to think of it, both has their advantages. Not having code duplication is definitely desirable but at the same time, importing methods from 3 different methods makes it difficult for someone to copy-paste the base method and use it on an actual CP problem.

Will resolve once you clear it up and we can come to a common ground :-)

Reply:
We always reimplement in each file to make copy-pasting during a contest easier. I meant using the code from that file rather than a recursive version which would be slower. Thanks

Reply:
Makes sense. Updated.